### PR TITLE
initial support for nrf9160

### PR DESCRIPTION
**Pull request by raiden00pl**
## Summary

- arch/arm: initial support for NRF91
    Port based on arch/arm/nrf53.
    Modem not fully supported yet. At the moment, initialization and AT interface work.
    Sockets and GNSS interface will be added later.
    Modem works only in a non-secure environment, so it's not easy to test it at the moment (need to switch from secure env to non-secure env).

- boards: initial support for nrf9160-dk

- Documentation: initial doc for NRF91

## Testing
nrf9160-dk

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/arm/src/nrf91/Kconfig [hunk 1]
```diff
@@ -0,0 +1,689 @@
...
+
+# NRF91 Families
+
+choice
+	prompt "NRF91 Chip Selection"
+	default ARCH_CHIP_NRF9160
+	depends on ARCH_CHIP_NRF91
+
+config ARCH_CHIP_NRF9160
+	bool "NRF9160"
```
Comment (acassis) [inline arch/arm/src/nrf91/Kconfig:16]:
```suggestion
	bool "nRF9160"

File: arch/arm/src/nrf91/Kconfig [hunk 2]
```diff
@@ -0,0 +1,689 @@
...
+
+config NRF91_SERIAL3
+	bool
+	default n
+
+config NRF91_SERIAL4
+	bool
+	default n
+
+menu "NRF91 Peripheral Selection"
```
Comment (acassis) [inline arch/arm/src/nrf91/Kconfig:69]:
```suggestion
menu "nRF91 Peripheral Selection"

**Comment by acassis** [comment]
According you Nordic website: 
https://www.nordicsemi.com/products/nrf9160

**Comment by raiden00pl** [comment]
@acassis  At the moment we use `NRF` for other Nordic chips, so it'll be better to change it to `nRF` in a separate PR and update all Nordic chips at once.

**Pull request merged**